dev/tools/dartdoc.dart

```diff
@@ -468,6 +468,8 @@ List<Directory> findPackages() {
       if (entity is! Directory)
         return false;
       final File pubspec = File('${entity.path}/pubspec.yaml');
+      if (!pubspec.existsSync())
+        return false;
```

Comment:
This is unlikely to actually be a problem, but nit: we should just `try` and `catch` a `FileSystemException`, returning false if we get one.

Reply:
I ran into this error because `_flutter_web_build_script` doens't have a pubspec.yaml, preventing me from running `dev/bots/docs.sh` locally.

Reply:
I'd rather check if it exists, since if a file exists, but we get an error trying to read it, I want to surface such an error rather than masking it.

Reply:
Ahh ok. In that case, can we print a warning to the developer that they have a bad/old folder in their repo?

We did something similar for this in https://github.com/flutter/flutter/pull/71950, where I ran into a similar problem for the same folder.

Reply:
sg - done.